### PR TITLE
add @wallet-standard/errors package

### PR DESCRIPTION
**Pull request by olistic**
This PR adds a handful of errors to be used by wallets and dApps. This would allow dApps to do better error-handling.

### CODE REVIEW

File: packages/core/standard/src/errors.ts [hunk 1]
```diff
@@ -1,2 +0,0 @@
-// TODO: add error codes
-export {};
```
Comment (olistic) [inline packages/core/standard/src/errors.ts:2]: @jordansexton I created a separate `@wallet-standard/errors` package. Let me know if you prefer for the errors to live here instead.

Comment (jordansexton) [reply packages/core/standard/src/errors.ts:2]: I think separate package is great

**Review by jordansexton — COMMENTED** [review]
My initial take, not firmly attached to this --

The general structure looks good. But we should avoid numeric codes that mean nothing, are not human readable, and the app has to map to something be useful (or worse, attempt and fail with instanceof checks). Using negative numbers for error codes is also really odd.

What about having a `type` field with names like `standard:<ErrorName>` and having all the errors extend a `StandardWalletError` class that implements a `WalletError` interface describing the added field as `type: IdentifierString` or something?

Wallets can implement their own custom errors just like they do with features, etc.

File: packages/core/standard/src/errors.ts (hunk 1 above)
Comment (tam6868) [reply packages/core/standard/src/errors.ts:2]: Hữu jelmy

**Comment by olistic** [comment]
@jordaaash @steveluscher what do we want to do about this? Is the comment above still current? If so, I'll respond and try to move this forward.

**Comment by steveluscher** [comment]
This is the way we've gone with errors. Please feel free to add to it if you need more codes! https://github.com/wallet-standard/wallet-standard/tree/master/packages/core/errors

The principle at work there is that errors should _not_ be implementations (bundled in your app, though encountered _extremely_ rarely) but rather types. In production mode, _all_ error messages get reaped from the bundle, and are reconstitutable offline.

**Pull request closed without merging**